app/controllers/spree/admin/payment_methods_controller_decorator.rb

```diff
@@ -57,12 +57,19 @@ def load_data
         else
           @providers = Gateway.providers.reject{ |p| p.name.include? "Bogus" }.sort{|p1, p2| p1.name <=> p2.name }
         end
+        @providers.reject!{ |p| p.name.ends_with? "StripeConnect" } unless show_stripe?
         @calculators = PaymentMethod.calculators.sort_by(&:name)
       end
 
       def load_hubs
         @hubs = Enterprise.managed_by(spree_current_user).is_distributor.sort_by!{ |d| [(@payment_method.has_distributor? d) ? 0 : 1, d.name] }
       end
+
+      # Show Stripe as an option if enabled, or if the
+      # current payment_method is already a Stripe method
+      def show_stripe?
+        Spree::Config.stripe_connect_enabled || @payment_method.try(:type) == "Spree::Gateway::StripeConnect"
+      end
```

Comment:
⚠️ DISCLAIMER: buzz words ahead ⚠️ 

I think there is room here to apply some "tell don't ask", instead of asking all the time if we support Stripe we should just get that info from some class defined starting up the app (strategy pattern maybe?). In that way you'll not have to remember to check that "if Stripe" in the future avoiding bugs and letting the code be more clear and easier to test.

BTW, this is not blocking of course. Just something that would be nice to start seeing in the codebase.

Reply:
The value of `Spree::Config.stripe_connect_enabled` is dynamic (can be changed by super admin), so we can't cache the value on app initialisation. Is that what you mean?

Reply:
Resolved?